### ISSUE TITLE
Issue: cmd line title on elevation / deelevation

### ISSUE THREAD

**Issue by highend** [gerardog/gsudo#91]
Hi,

Windows 2019 v10.0.17763.107 here with gsudo v0.7.3

After opening a normal non-elevated command prompt and executing
`gsudo`
The title of the command prompt changes from
`C:\Windows\System32\cmd.exe`
to
`Administrator: C:\Windows\System32\cmd.exe - gsudo`

When I now remove the elevation with:
`gsudo -i medium`

The new title is:
`Administrator: C:\Windows\System32\cmd.exe - gsudo - gsudo -i medium`

01. Shouldn't it remove the "Administrator" at the beginning?
It's a non-evelated cmd now...

02. Can you clean up the right part of the title instead of appending everything?

Thanks!

**Comment by gerardog** [comment]
Please consider that, up to now, `gsudo` does not alter the console title in any way. AFAIK it is `cmd.exe` and Powershell who put the `Administrator:` prefix and on `cmd`, appends each child process name (invoked via CMD) as in `Process1.exe - Prcoess2.exe /param1'. Install and try 'yori' shell and you can see how different from CMD is regarding this.

Point # 1 could be done, as a workaround for cmd never removing the prefix when a non-admin cmd is called from an admin cmd. That means your example would be `C:\Windows\System32\cmd.exe - gsudo - gsudo -i medium` with the `Administrator:` prefix removed. Nonetheless lets find a way to figure out all possible localizations of "Administrator:" prefix, in any language and/or windows version like spanish or chinese. “管理员：命令提示符”. Maybe removing everything before the colon char? (on de-elevations only).

![image](https://user-images.githubusercontent.com/3901474/126883385-4cd738d3-219d-41e3-b018-d9d86866cd92.png)
![image](https://user-images.githubusercontent.com/3901474/126883539-42f231de-addb-4986-9b0d-9e410eb40144.png)

Regarding # 2, I am not sure I understand what you mean. In your example, after fixing # 1 the console title should behave in the following way.

Run CMD -> `C:\Windows\System32\cmd.exe`
Elevate with gsudo -> `Administrator: C:\Windows\System32\cmd.exe - gsudo`
Deelevate -> `C:\Windows\System32\cmd.exe - gsudo - gsudo -i medium`

... because, I rather alter as little as possible how a cmd/powershell works. Also if my call stack has a de-elevation I prefer to notice by looking at the title.

**Comment by gerardog** [comment]
De-elevations will remove whatever is left to the `:`.

If you try to run these tests from an **elevated** cmd (with a [CI build](https://ci.appveyor.com/project/gerardog/gsudo/builds/40206447/artifacts) of gsudo or the upcoming v0.8) ...
```
gsudo -i medium powershell
gsudo -i low powershell
gsudo -i medium nslookup
gsudo -i low nslookup
gsudo -i medium cmd
gsudo -i low cmd
```
... you will see the Admin prefix is removed, except in the last case...

It seems there is a glitch somewhere for CMD for tokens generated with the Safer Api. (gsudo uses it for low integrity tokens only). Even if gsudo removes the prefix, the new low-integrity non-admin CMD adds it again. Not sure why.

**Comment by highend** [comment]
Yeah, tried the `CI build` and it works fine so far (except the last case as you mentioned). Thanks for implementing this!

**Comment by gerardog** [comment]
Released in [v1.0.0](https://github.com/gerardog/gsudo/releases/tag/v1.0.0) (PR #94)
